src/mrdog/config_storage.rs

```diff
@@ -1,5 +1,93 @@
-use super::ConfigItem;
+use dirs;
+use serde::{Deserialize, Serialize};
+use std::{
+    fs, io,
+    path::{self, PathBuf},
+};
 
-pub fn set_config(item: ConfigItem) {
-    todo!("Set config {item:?}")
+use super::{ConfigValue, Provider};
+
+#[derive(Debug)]
+pub enum ConfigError {
+    CreateFailed(io::Error),
+    CouldNotConvertToToml(toml::ser::Error),
+    WriteToConfigFailed(io::Error),
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+struct Config {
```

Comment:
Now that I see it, adding "version" might be a good idea when someone starts using except myself and my mom.